config/v1/types_apiserver.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// Default references a secret containing the default TLS cert info for serving secure traffic.
+	Default corev1.SecretReference `json:"default,omitempty"`
+	// Named references secrets containing the TLS cert info for serving secure traffic to specific hostnames.
+	Named []APIServerNamedServingCert `json:"named,omitempty"`
+}
+
+type APIServerNamedServingCert struct {
+	// Names is a list of DNS names this certificate should be used to secure
+	// A name can be a normal DNS name, or can contain leading wildcard segments.
+	Names []string `json:"names"`
+	// Secret references the secret containing the TLS cert info for serving secure traffic.
+	Secret corev1.SecretReference `json:"secret"`
+}
+
+type APIServerStatus struct {
```

Comment:
@deads2k @sanchezl do we want to report anything here? or just have to be here for consistency?

Reply:
> @deads2k @sanchezl do we want to report anything here? or just have to be here for consistency?

consistency for now